### PR TITLE
refactor: Simplify beforeRunTests return singature

### PR DESCRIPTION
**Pull request by jan-gogo**
This PR is a cherrypicked refactor from #1321 

After merging this one, the #1321 will slim down a little.

## Checklist

- [x] Simplify beforeRunTests return signature
- [x] Treat Args as context

### CODE REVIEW

**Review by Sloox — APPROVED** [review]

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt [hunk 1]
```diff
@@ -12,31 +13,31 @@ import ftl.util.obfuscatePrettyPrinter
 import java.nio.file.Files
 import java.nio.file.Paths
 
-suspend fun dumpShards(
-    args: AndroidArgs,
+suspend fun AndroidArgs.dumpShards(
+    @VisibleForTesting
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/run/DumpShards.kt:17]: `@VisibleForTesting` is used for field, class, method visibility, not for a method parameter, which do not have any visibility. We should not use it

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt [hunk 2]
```diff
@@ -12,31 +13,31 @@ import ftl.util.obfuscatePrettyPrinter
...
         size = shards.flatMap { it.value.shards.values }.count(),
-        obfuscatedOutput = args.obfuscateDumpShards
+        obfuscatedOutput = obfuscateDumpShards
     )
 }
 
-fun dumpShards(
-    args: IosArgs,
+fun IosArgs.dumpShards(
+    @VisibleForTesting
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/run/DumpShards.kt:33]: same here

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt [hunk 3]
```diff
@@ -12,31 +13,31 @@ import ftl.util.obfuscatePrettyPrinter
...
     )
-    val shards: AndroidMatrixTestShards = args.getAndroidMatrixShards()
+    val shards: AndroidMatrixTestShards = getAndroidMatrixShards()
     saveShardChunks(
         shardFilePath = shardFilePath,
         shards = shards,
         size = shards.flatMap { it.value.shards.values }.count(),
-        obfuscatedOutput = args.obfuscateDumpShards
+        obfuscatedOutput = obfuscateDumpShards
     )
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/run/DumpShards.kt:29]: How about?

```kotlin
getAndroidMatrixShards().run {
    saveShardChunks(
        shardFilePath = shardFilePath,
        shards = this,
        size = flatMap { it.value.shards.values }.count(),
       obfuscatedOutput = obfuscateDumpShards
    )
}
```

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt [hunk 4]
```diff
@@ -12,31 +13,31 @@ import ftl.util.obfuscatePrettyPrinter
...
         "Cannot dump shards for non instrumentation test, ensure test apk has been set."
     )
-    val shards: AndroidMatrixTestShards = args.getAndroidMatrixShards()
+    val shards: AndroidMatrixTestShards = getAndroidMatrixShards()
     saveShardChunks(
         shardFilePath = shardFilePath,
         shards = shards,
         size = shards.flatMap { it.value.shards.values }.count(),
-        obfuscatedOutput = args.obfuscateDumpShards
+        obfuscatedOutput = obfuscateDumpShards
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/run/DumpShards.kt:28]: How about

```kotlin
getAndroidMatrixShards().run {
    saveShardChunks(
        shardFilePath = shardFilePath,
        shards = this,
        size = flatMap { it.value.shards.values }.count(),
       obfuscatedOutput = obfuscateDumpShards
    )
}
```

?

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt [hunk 5]
```diff
@@ -12,31 +13,31 @@ import ftl.util.obfuscatePrettyPrinter
...
     )
 }
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/run/DumpShards.kt:30]: How about

```kotlin
getAndroidMatrixShards().run {
    saveShardChunks(
        shardFilePath = shardFilePath,
        shards = this,
        size = flatMap { it.value.shards.values }.count(),
       obfuscatedOutput = obfuscateDumpShards
    )
}
```

?

File: test_runner/src/main/kotlin/ftl/run/NewTestRun.kt [hunk 6]
```diff
@@ -18,29 +18,30 @@ import ftl.run.platform.runIosTests
 import kotlinx.coroutines.TimeoutCancellationException
 import kotlinx.coroutines.withTimeoutOrNull
 
-suspend fun newTestRun(args: IArgs) = withTimeoutOrNull(args.parsedTimeout) {
+suspend fun IArgs.newTestRun() = withTimeoutOrNull(parsedTimeout) {
+    val args: IArgs = this@newTestRun
     println(args)
-    val (matrixMap, testShardChunks, ignoredTests) = cancelTestsOnTimeout(args.project) { runTests(args) }
+    val (matrixMap, testShardChunks, ignoredTests) =
+        cancelTestsOnTimeout(project) { runTests() }
```
Comment (piotradamczyk5) [inline test_runner/src/main/kotlin/ftl/run/NewTestRun.kt:25]: could be single line

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt (hunk 1 above)
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/DumpShards.kt:17]: For me, it's the only convention for marking why something is exposed but is not used in production code.

Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/run/DumpShards.kt:17]:
```suggestion
    // VisibleForTesting
```

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt (hunk 2 above)
Comment (jan-gogo) [inline test_runner/src/main/kotlin/ftl/run/DumpShards.kt:33]:
```suggestion
    // VisibleForTesting
```

File: test_runner/src/main/kotlin/ftl/run/NewTestRun.kt (hunk 6 above)
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/NewTestRun.kt:25]: single line is not fully visible on 14 cal split screen ;P.

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt (hunk 5 above)
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/DumpShards.kt:30]: I don't like using `val` but also I don't like unnecessary nested scopes and explicit `this` keyword so it is the same for me but the current version gives fewer changes in diff.

File: test_runner/src/main/kotlin/ftl/run/NewTestRun.kt (hunk 6 above)
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/NewTestRun.kt:25]: single line is not fully visible on 14 cal split screen ;P.

File: test_runner/src/main/kotlin/ftl/run/DumpShards.kt (hunk 5 above)
Comment (jan-gogo) [reply test_runner/src/main/kotlin/ftl/run/DumpShards.kt:30]: I don't like using val but also I don't like unnecessary nested scopes and explicit this keyword so it is the same for me but the current version gives fewer changes in diff.

**Review by piotradamczyk5 — APPROVED** [review]

**Pull request merged**